src/util/octokit.ts

```diff
@@ -18,9 +18,19 @@ export const paginate = (
   while (octokit.hasNextPage(response)) {
     // Avoid GitHub's "abuse detection mechanisms"
     await wait(2000);
-
-    response = await octokit.getNextPage(response); // eslint-disable-line no-await-in-loop
-    data = data.concat(extractItems(response.data));
+    try {
+      response = await octokit.getNextPage(response); // eslint-disable-line no-await-in-loop
+      data = data.concat(extractItems(response.data));
+    } catch (e) {
+      if (e.headers && e.headers['retry-after']) {
+        const retryAfter = e.headers['retry-after'];
+        console.info(`\nHit rate limit; waiting ${retryAfter} seconds and retrying.`);
```

Comment:
Linter is erroring on `console.info()` here but I want to give the user feedback when this happens (since it likely means +60s each time). What is the recommended approach?

Reply:
We'll want to use the `logger` object like we do here: https://github.com/NerdWalletOSS/shepherd/blob/a5a092f2342f40dc20272eeee174a01d3332c3cb/src/commands/checkout.ts#L52

We don't have access to that in this helper. I think the best way to do this is to pass a `onRetry` callback from `commands/checkout.ts` through to `adapters/github.ts` all the way down to this helper so we can notify the cli command layer (which is responsible for input/output) about when a retry is happening.

Reply:
Instead of passing an `onRetry`, I imported `logger` directly. Thoughts?

Reply:
Orrr typescript doesn't like that for some reason.

Reply:
I'm not completely opposed, but the thought was to keep the adapters and utilities clean of logging so they could potentially be used in other contexts than the CLI.